### ISSUE TITLE
Stack overflow since version 1.0.0-alpha3

### ISSUE THREAD

**Issue by kateinoigakukun** [wasmerio/wasmer#1674]
### Describe the bug

Binary: https://gist.github.com/kateinoigakukun/bb62851920f5935d6f23fc9a88fff022

This wasm binary can't be run with wasmer 1.0.0-alpha3 with cranelift backend.

```
wasmer 1.0.0-alpha3 | rustc 1.42.0-nightly (0de96d37f 2019-12-19) | x86_64
```

### Steps to reproduce

```sh
$ wget https://gist.github.com/kateinoigakukun/bb62851920f5935d6f23fc9a88fff022/raw/cb41560e14dfa7f05ff49fb20f1ac6c1c1013d5e/a.out
$ wasmer run a.out --llvm
[ RUN      ] NumericParsing.UInt8/success
[       OK ] NumericParsing.UInt8/success
[ RUN      ] NumericParsing.Int8/success
[       OK ] NumericParsing.Int8/success
[ RUN      ] NumericParsing.UInt16/success
[       OK ] NumericParsing.UInt16/success
[ RUN      ] NumericParsing.Int16/success
[       OK ] NumericParsing.Int16/success
[ RUN      ] NumericParsing.UInt32/success
[       OK ] NumericParsing.UInt32/success
[ RUN      ] NumericParsing.Int32/success
[       OK ] NumericParsing.Int32/success
[ RUN      ] NumericParsing.UInt64/success
[       OK ] NumericParsing.UInt64/success
[ RUN      ] NumericParsing.Int64/success
[       OK ] NumericParsing.Int64/success
[ RUN      ] NumericParsing.UInt/success
[       OK ] NumericParsing.UInt/success
[ RUN      ] NumericParsing.Int/success
[       OK ] NumericParsing.Int/success
[ RUN      ] NumericParsing.FixedWidthInteger/maxUInt64
[       OK ] NumericParsing.FixedWidthInteger/maxUInt64
[ RUN      ] NumericParsing.Float/Basics
[       OK ] NumericParsing.Float/Basics
[ RUN      ] NumericParsing.Double/Basics
[       OK ] NumericParsing.Double/Basics
NumericParsing: All tests passed
$ wasmer run a.out --singlepass
# Same output as llvm backend
$ wasmer run a.out --cranelift
[ RUN      ] NumericParsing.UInt8/success

thread 'main' has overflowed its stack
fatal runtime error: stack overflow

```

### Expected behavior

No stack overflow exception

**Comment by nlewycky** [comment]
This was just fixed in #1671! We haven't had a release with it yet.

**Comment by kateinoigakukun** [comment]
Thank you for letting me know 😄 I'm looking forward to the next release.
